plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+        self.log('Getting Network Device information', "INFO")
+        try:
+            all_mac_address = [eachdevice["mac_address"] for eachdevice in self.payload["config"]]
+            device_list = self.dnac.devices.get_device_list(macAddress=all_mac_address)
+            if self.payload["device_fields"] == "" or self.payload["device_fields"] == "all":
+                self.payload["device_list"] = device_list['response']
+                return self.payload["device_list"]
+
+            fields = [str(self.keymap[x]) for x in self.payload["device_fields"].split(",")]
+            if len(device_list) != 0:
+                df = pd.DataFrame.from_records(device_list['response'])
+                selected_data = df[fields]
+                self.payload["device_list"] = selected_data.to_dict('records')
+                return self.payload["device_list"]
+            else:
+                return None
+        except Exception as e:
+            self.log("Unable to get device info "+ str(e) , "ERROR")
+
+    def get_ap_configuration(self):
+        """
+        This function used to get AP device details as json response from DNAC site.
+        by giving MAC address as a input in the URL GET Method
+        Parameters:
+            We are not using input param already passed in the module.param
+          - self.payload: used from yml input files input.yml we are using accesspoints section
+            if ap_selected_field in input.yml is empty or all this will show all field
+            else ap_selected_field: "mac_address,eth_mac,ap_name,led_status,location"
+            then will show only listed field.
+        Returns:
+            [
+                {
+                    "apName": "NFW-AP1-9130AXE",
+                    "ethMac": "34:5d:a8:0e:20:b4",
+                    "ledBrightnessLevel": 3,
+                    "ledStatus": "Enabled",
+                    "location": "LTTS",
+                    "macAddress": "90:e9:5e:03:f3:40"
+                }
+            ]
+        Example:
+            functions = Accesspoint(module)
+            ap_data = functions.get_ap_configuration()
+        """
+        ap_config_data = []
+        for device in self.payload["device_list"]:
+            self.log('Getting Access Point Configuration Information for ' \
+                     + device['apEthernetMacAddress'], "INFO")
+            try:
+                jsondata = self.dnac.wireless.get_access_point_configuration(
+                    key=str(device['apEthernetMacAddress']))
+                ap_config_data.append(jsondata)
+            except Exception as e:
+                self.log(jsondata['error'] + e, "ERROR")
+        self.log('Access Point Configuration Information: ' + str(ap_config_data), "INFO")
+        if self.payload["ap_selected_field"] == "" or self.payload["ap_selected_field"] == "all":
+            return ap_config_data
+
+        fields = [str(self.keymap[x]) for x in self.payload["ap_selected_field"].split(",")]
```

Comment:
Don't do multiple operation in single line.

Reply:
modified